pkg/controller/installation/core_controller.go

```diff
@@ -689,6 +717,15 @@ func GenerateRenderConfig(install *operator.Installation) render.NetworkConfig {
 	// If CalicoNetwork is specified, then use Calico networking.
 	if install.Spec.CalicoNetwork != nil {
 		config.CNI = render.CNICalico
+		if install.Spec.CalicoNetwork.BPFDataplaneMode != nil {
```

Comment:
I don't think we need this extra abstraction - the `Installation` object is passed through to render so we can just use it directly there.

I understand that there is already some abstraction here and it's inconsistent, but the vast majority of fields are passed straight through via the Installation object into the `render` code, and this is the exception case.

Reply:
sure, I am all for that, was wondering why we rewrite one thing into another, but felt obliged to follow the pattern :-)